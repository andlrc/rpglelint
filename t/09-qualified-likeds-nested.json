[
   {
      "column":8,
      "file":"t/09-qualified-likeds-nested.rpgle",
      "line":2,
      "message":"data structure 'not_qual' needs to be qualified",
      "severity":"warning"
   },
   {
      "column":10,
      "file":"t/09-qualified-likeds-nested.rpgle",
      "line":13,
      "message":"data structure 'lnot_qual' needs to be qualified",
      "severity":"warning"
   },
   {
      "column":10,
      "file":"t/09-qualified-likeds-nested.rpgle",
      "line":14,
      "message":"data structure 'llnot_qual' needs to be qualified",
      "severity":"warning"
   },
   {
      "column":10,
      "file":"t/09-qualified-likeds-nested.rpgle",
      "line":18,
      "message":"data structure 'not_qual2' needs to be qualified",
      "severity":"warning"
   },
   {
      "column":10,
      "file":"t/09-qualified-likeds-nested.rpgle",
      "line":28,
      "message":"data structure 'lnot_qual2' needs to be qualified",
      "severity":"warning"
   },
   {
      "column":10,
      "file":"t/09-qualified-likeds-nested.rpgle",
      "line":29,
      "message":"data structure 'llnot_qual2' needs to be qualified",
      "severity":"warning"
   }
]
