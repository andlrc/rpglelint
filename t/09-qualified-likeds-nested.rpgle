// rpglelint: -Wqualified
dcl-ds not_qual;
  sub1 int(10);
  sub2 int(20);
end-ds;

dcl-ds qual qualified;
  sub1 int(10);
  sub2 int(20);
end-ds;

dcl-proc my_proc;
  dcl-ds lnot_qual likeds(not_qual);
  dcl-ds llnot_qual likeds(lnot_qual);
  dcl-ds lqual likeds(qual);
  dcl-ds llqual likeds(lqual);

  dcl-ds not_qual2;
    sub1 int(10);
    sub2 int(20);
  end-ds;

  dcl-ds qual2 qualified;
    sub1 int(10);
    sub2 int(20);
  end-ds;

  dcl-ds lnot_qual2 likeds(not_qual2);
  dcl-ds llnot_qual2 likeds(lnot_qual2);
  dcl-ds lqual2 likeds(qual2);
  dcl-ds l_qual2 likeds(lqual2);
end-proc;
