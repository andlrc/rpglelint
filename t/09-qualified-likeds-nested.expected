t/09-qualified-likeds-nested.rpgle:2:8: warning: data structure 'not_qual' needs to be qualified [-Wqualified]
 dcl-ds not_qual;
        ^
t/09-qualified-likeds-nested.rpgle:13:10: warning: data structure 'lnot_qual' needs to be qualified [-Wqualified]
   dcl-ds lnot_qual likeds(not_qual);
          ^
t/09-qualified-likeds-nested.rpgle:14:10: warning: data structure 'llnot_qual' needs to be qualified [-Wqualified]
   dcl-ds llnot_qual likeds(lnot_qual);
          ^
t/09-qualified-likeds-nested.rpgle:18:10: warning: data structure 'not_qual2' needs to be qualified [-Wqualified]
   dcl-ds not_qual2;
          ^
t/09-qualified-likeds-nested.rpgle:28:10: warning: data structure 'lnot_qual2' needs to be qualified [-Wqualified]
   dcl-ds lnot_qual2 likeds(not_qual2);
          ^
t/09-qualified-likeds-nested.rpgle:29:10: warning: data structure 'llnot_qual2' needs to be qualified [-Wqualified]
   dcl-ds llnot_qual2 likeds(lnot_qual2);
          ^
